Add boilerplate media type definitions for each protocol message.

chris-wood opened the pull request:
Closes #105

chris-wood commented:
@mnot -- is it better practice to define media types per protocol message, or to define a single media type for all protocol messages?

mnot commented:
Generally, if they're different formats -- e.g., they have different schema, or just different syntax -- they need different media types (or an explicitly generic one like `application/octet-stream`). If it's one format that has different uses, you can give it a single media type and then switch contexts based upon internal flags.

The real question, though, is whether you have any potential for use cases where it's important for generic software that's handling these messages to do different things based upon the media type.

chris-wood commented:
> Generally, if they're different formats -- e.g., they have different schema, or just different syntax -- they need different media types (or an explicitly generic one like application/octet-stream). If it's one format that has different uses, you can give it a single media type and then switch contexts based upon internal flags.
>
> The real question, though, is whether you have any potential for use cases where it's important for generic software that's handling these messages to do different things based upon the media type.

I don't think there'd ever be generic software using these types (beyond the HpkeConfig?). Most messages are meant to be consumed by PPM and PPM alone. I moved protocol messages to the "message" registry space and kept the HpkeConfig in the application type space, and also left all intended usage fields as COMMON. I also added an open issue to have someone more familiar with these registries give these allocation requests a review.

@ekr, this should be good to go based on the above.

ekr commented on draft-pda-protocol.md:
```diff
@@ -495,7 +495,7 @@ keys for a reasonable period after key changes in order to avoid
 [[OPEN ISSUE: https://github.com/abetterinternet/prio-documents/issues/106]]
 
-### Upload Request
+### Report
```
This hed seems a bit short. What's your thinking here.

chris-wood replied:
Only that it was more clear, but it's a separate change, so I'll just revert.

ekr reviewed (approved):
LGTM except for the hed.
